src/components/Spacing/Spacing.js

```diff
@@ -0,0 +1,50 @@
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+const baseStyles = () => css`
+  label: spacing;
+`;
+
+const marginBottomStyles = ({ theme, bottom }) =>
+  bottom &&
+  css`
+    label: spacing--margin-bottom;
+    margin-bottom: ${theme.spacings.giga};
+  `;
+
+const marginTopStyles = ({ theme, top }) =>
+  top &&
+  css`
+    label: spacing--margin-top;
+    margin-top: ${theme.spacings.giga};
+  `;
+
+/**
+ * Margin helper
+ */
+const Spacing = styled('div')`
+  ${baseStyles}
+  ${marginBottomStyles}
+  ${marginTopStyles}
+`;
+
+Spacing.propTypes = {
+  /**
+   * Adds bottom margin to component
+   */
+  bottom: PropTypes.bool,
+  /**
+   * Adds top margin to component
+   */
+  top: PropTypes.bool
+};
+
+Spacing.defaultProps = {
+  bottom: false,
+  top: false
+};
+
+/**
+ * @component
+ */
+export default Spacing;
```

Comment:
Most of our components have default bottom spacing. Can you give an example when this component would be used (instead)?

Reply:
This is a initial discussion we had yesterday. This approach is meant to remove default spacing from components and allow the developer to use dynamic spacing, instead of having every layout component with the `noMargin` option you wrap it in the `Spacing` component. For example: https://github.com/sumup/ze-dashboard/pull/1576/files#diff-1789329109b953ddc27bc81f108826faR22

I understand it creates a bit more of code when doing it, but the idea is to move the layout/position responsibilities from the components and move it to layout wrappers

Reply:
<Spacing /> component sounds great! There are often cases when you have component composition where is unclear which component should be the one with the margin from the other components and can be just a nuisance to deal with.

May be the bottom/top props can receive a `size` attribute for different uses, but this can be added later in case we need it?

Reply:
@elanoism that's exactly what I thought. Let's start with the gutter then add dynamic values as we go.

Reply:
Haha, good to see you had this discussion. I was reading top to bottom 😆